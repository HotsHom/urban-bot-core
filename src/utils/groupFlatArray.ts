export function groupFlatArray<T>(arr: T[], groupLength: number): T[][] {
    if (groupLength < 1) {
        throw new Error(`groupLength should be more than zero`);
    }

    const groupArray = [];

    for (let startSliceIndex = 0, groupArrayIndex = 0; startSliceIndex < arr.length; startSliceIndex += groupLength) {
        groupArray[groupArrayIndex] = arr.slice(startSliceIndex, startSliceIndex + groupLength);

        groupArrayIndex += 1;
    }

    return groupArray;
}
