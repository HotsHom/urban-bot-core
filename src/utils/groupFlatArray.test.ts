import { groupFlatArray } from './groupFlatArray';

describe('groupFlatArray', () => {
    it("can't pass zero", () => {
        expect(() => groupFlatArray([1, 2, 3], 0)).toThrowErrorMatchingSnapshot();
    });

    it("can't pass less than zero", () => {
        expect(() => groupFlatArray([1, 2, 3], -1)).toThrowErrorMatchingSnapshot();
    });

    it('return empty array', () => {
        expect(groupFlatArray([], 1)).toEqual([]);
    });

    it('return group array with one element', () => {
        expect(groupFlatArray([1], 1)).toEqual([[1]]);
        expect(groupFlatArray([1], 2)).toEqual([[1]]);
    });

    it('return group array with two elements', () => {
        expect(groupFlatArray([1, 2], 1)).toEqual([[1], [2]]);
        expect(groupFlatArray([1, 2], 2)).toEqual([[1, 2]]);
    });

    it('return group array with three elements', () => {
        expect(groupFlatArray([1, 2, 3], 1)).toEqual([[1], [2], [3]]);
        expect(groupFlatArray([1, 2, 3], 2)).toEqual([[1, 2], [3]]);
        expect(groupFlatArray([1, 2, 3], 3)).toEqual([[1, 2, 3]]);
    });

    it('return group array with four elements', () => {
        expect(groupFlatArray([1, 2, 3, 4], 2)).toEqual([
            [1, 2],
            [3, 4],
        ]);
        expect(groupFlatArray([1, 2, 3, 4], 3)).toEqual([[1, 2, 3], [4]]);
        expect(groupFlatArray([1, 2, 3, 4], 5)).toEqual([[1, 2, 3, 4]]);
    });
});
