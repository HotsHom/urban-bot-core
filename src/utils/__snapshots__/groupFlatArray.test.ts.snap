// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`groupFlatArray can't pass less than zero 1`] = `"groupLength should be more than zero"`;

exports[`groupFlatArray can't pass zero 1`] = `"groupLength should be more than zero"`;
